tech_demos/loader_animations/pure_css_stars_bevel.html

```diff
@@ -58,57 +66,147 @@
       star_29_s_notrail: 64,
     };
 
-    $('#logo_svg').ready(function(){
-
-    var percentage = 0;
-    var anim_timer = null;
-    var animation_cycler = function () {
-      var svg = document.getElementById('logo_svg').contentDocument;
+var current_percent = 0;
+function animation_cycler()
+{
+  var svg = document.getElementById('logo_svg').contentDocument;
 
-      if (svg == null || typeof svg.getElementById != 'function') {
-        return false;
-      }
+  if (svg == null || typeof svg.getElementById != 'function') {
+    return false;
+  }
 
-      if (percentage < global_percent) {
-        for (var id in star_animation_timing) {
-          if (star_animation_timing[id] <= global_percent) {
-            var star = svg.getElementById(id);
+  if (current_percent < global_percent) {
+    for (var id in star_animation_timing) {
+      if (star_animation_timing[id] <= global_percent) {
+        var star = svg.getElementById(id);
 
-            star.style.animationPlayState = 'running';
-            console.max(id, star.style.animationPlayState);
-          }
-          percentage = global_percent;
-        }
+        star.style.animationPlayState = 'running';
+        console.max(id, star.style.animationPlayState);
       }
-      else if (percentage >= 100) {
-        svg.getElementById('flare_top').style.display = 'block';
-        svg.getElementById('flare_bottom').style.display = 'block';
-        clearInterval(anim_timer);
+      current_percent = global_percent;
+    }
+  }
+  else if (current_percent >= 100) {
+    // Wait until the stars have all finished before we animate the flare!
+    for (var id in star_animation_timing) {
+      var star = svg.getElementById(id);
+      if (star.style.animationPlayState == 'running') {
+//	console.max("oi!", "waiting for stars to finish before doing flare");
```

Comment:
Was this unfinished?

Reply:
Yes, it's supposed to stop the flares appearing until all the stars have fully animated, but I couldn't immediately see a way to do this as animationPlayState stays at "running" and I haven't looked into it further yet, so it's commented out and for the time being the flares just appear at 100%.

Reply:
We may have to listen for the animationend event for each star?

Reply:
Yes if such a thing exists and triggers on these animations. Like I say I didn't have time to research it yet. num_animations_running++ when an animation is triggered and simultaneously add an event listener that does num_animations_running-- would be fine.

This is just to stop it looking weird during a fast load when the flares appear before the stars that supposedly have flare on them are actually there.